Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs

```diff
@@ -450,4 +451,30 @@ private async Task SaveParticipantScore(Submission submission)
             submission.ProcessingComment = $"Exception in SaveParticipantScore: {ex.Message}";
         }
     }
+
+    private async Task<IEnumerable<SubmissionResultsServiceModel>> GetUserSubmissions(
+        int problemId,
+        bool isOfficial,
+        int take,
+        UserInfoModel user,
+        Problem? problem,
+        Participant? participant)
+    {
+        var validationResult = this.submissionResultsValidationService.GetValidationResult((user, problem, participant, isOfficial));
+        if (!validationResult.IsValid)
+        {
+            throw new BusinessServiceException(validationResult.Message);
+        }
+
+        var userSubmissions = this.submissionsData
+            .GetAllByProblemAndParticipant(problemId, participant!.Id)
+            .MapCollection<SubmissionResultsServiceModel>();
+
+        if (take != 0)
```

Comment:
Move this take before the `.MapCollection<SubmissionResultsServiceModel>();` invocation. This way you will save up operations on unnesessary records that are being retrieved from the db.